iceoryx_posh/test/moduletests/test_popo_typed_publisher.cpp

```diff
@@ -45,32 +55,77 @@ class TypedPublisherTest : public Test
 
   protected:
     ChunkMock<DummyData> chunkMock;
-    std::function<void(DummyData* const)> placeholderDeleter = [](DummyData* const) {};
     TestTypedPublisher sut{{"", "", ""}};
+    MockPublisherPortUser& portMock{sut.mockPort()};
 };
 
-TEST_F(TypedPublisherTest, LoansSamplesLargeEnoughForTheType)
+
+TEST_F(TypedPublisherTest, LoansChunksLargeEnoughForTheType)
 {
-    // ===== Setup ===== //
-    auto sample = iox::popo::Sample<DummyData>(
-        {static_cast<DummyData*>(chunkMock.chunkHeader()->payload()), placeholderDeleter}, sut);
-    EXPECT_CALL(sut, loan(sizeof(DummyData)))
-        .WillOnce(Return(ByMove(iox::cxx::success<iox::popo::Sample<DummyData>>(std::move(sample)))));
+    EXPECT_CALL(portMock, tryAllocateChunk(sizeof(DummyData)))
+        .WillOnce(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
+    // ===== Test ===== //
+    auto result = sut.loan();
+    // ===== Verify ===== //
+    EXPECT_FALSE(result.has_error());
+    EXPECT_CALL(portMock, releaseChunk(chunkMock.chunkHeader()));
+    // ===== Cleanup ===== //
+}
 
+TEST_F(TypedPublisherTest, LoanedSampleIsDefaultInitialized)
+{
+    EXPECT_CALL(portMock, tryAllocateChunk(sizeof(DummyData)))
+        .WillOnce(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
     // ===== Test ===== //
     auto result = sut.loan();
     // ===== Verify ===== //
-    EXPECT_EQ(false, result.has_error());
+    ASSERT_FALSE(result.has_error());
+    EXPECT_EQ(result.value()->val, DummyData::defaultVal());
+    EXPECT_CALL(portMock, releaseChunk(chunkMock.chunkHeader()));
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedPublisherTest, LoanWithArgumentsCallsCustomCtor)
+{
+    constexpr uint64_t CUSTOM_VALUE{73};
+    EXPECT_CALL(portMock, tryAllocateChunk(sizeof(DummyData)))
+        .WillOnce(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
+    // ===== Test ===== //
+    auto result = sut.loan(CUSTOM_VALUE);
+    // ===== Verify ===== //
+    ASSERT_FALSE(result.has_error());
+    EXPECT_EQ(result.value()->val, CUSTOM_VALUE);
+    EXPECT_CALL(portMock, releaseChunk(chunkMock.chunkHeader()));
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedPublisherTest, LoanPreviousSampleSucceeds)
+{
+    EXPECT_CALL(portMock, tryGetPreviousChunk())
+        .WillOnce(Return(ByMove(iox::cxx::optional<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
+    // ===== Test ===== //
+    auto result = sut.loanPreviousSample();
+    // ===== Verify ===== //
+    EXPECT_TRUE(result.has_value());
+    EXPECT_CALL(portMock, releaseChunk(chunkMock.chunkHeader()));
+    // ===== Cleanup ===== //
+}
+
+TEST_F(TypedPublisherTest, LoanPreviousSampleFails)
```

Comment:
No it is just named differently because it does even a bit more ``LoanForwardsAllocationErrorsToCaller`` I renamed it to ``LoanFailsAndForwardsAllocationErrorsToCaller``. I would not have two tests for a) the failure and b) the forwarded error type, common sense and all that jazz.